dependencies.yaml

```diff
@@ -264,8 +287,17 @@ dependencies:
       - output_types: [conda, pyproject]
         packages:
           - &numpy numpy>=1.21
-          - *cuda_python
           - *rmm
+    specific:
+      - output_types: [conda, requirements, pyproject]
+        matrices:
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-python>=12.0,<13.0a0
```

Comment:
If you want, you can use YAML anchors. The first time this pinning is defined, write:
```yaml
- &cuda_python12 cuda-python>=12.0,<13.0a0
```
and later you can write
```yaml
- *cuda_python12
```